src/formatDate.js

```diff
@@ -8,7 +8,28 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const partsOfDate = date.split(oldSeparator);
+  const dateObject = {};
+
+  for (let i = 0; i < 3; i++) {
```

Comment:
The loop's purpose is clear, but it relies on a 'magic number' 3 which represents the expected number of date parts. It would be more maintainable to use a constant or derive this number from the length of `fromFormat` or `partsOfDate` arrays to avoid hardcoding values.